File: ApiAgendamento/Domain/Entities/Usuario.cs
namespace ApiAgendamento.Domain.Entities;

public class Usuario
{
    public int Id { get; set; }
    public string Email { get; set; } = string.Empty;
    public string SenhaHash { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty; // "medico" ou "paciente"
    public int? MedicoId { get; set; }
    public int? PacienteId { get; set; }
}

File: frontend-app/src/models/Usuario.ts
// Modelo para Usuario
export interface Usuario {
  id: number;
  nome: string;
  email: string;
  senha: string;
  tipo: 'paciente' | 'medico';
  medicoId?: number; // Opcional, se for paciente
  pacienteId?: number; // Opcional, se for médico
}
